### PR TITLE
ci: Combine actions to build Yocto

### PR DESCRIPTION
**Pull request by doanac**
Implements #24

### CODE REVIEW

**Comment by doanac** [comment]
This is sort of a chicken and egg. I think we have to merge it before we'll start seeing nightly runs.

File: .github/workflows/build-yocto.yml [hunk 1]
```diff
@@ -1,12 +1,13 @@
-name: Push Build
+name: Build Yocto
 
 on:
+  pull_request:
```
Comment (ricardosalveti) [inline .github/workflows/build-yocto.yml:4]: We had push as a separated file as we also want to implement https://github.com/quic-yocto/meta-qcom-hwe/issues/21, which needs a post build action of saving the yml lock file (similar to pinned manifest), and I believe we might want to save just for push (we could save for prs as well, but not sure if that is valuable).

We can combine now and split later if a new job turns out to be needed.

**Comment by doanac** [comment]
@ricardosalveti - i've set this PR to be the one cleanup. The nightly build PR is here: https://github.com/quic-yocto/meta-qcom-hwe/pull/26

File: .github/workflows/build-yocto.yml (hunk 1 above)
Comment (doanac) [reply .github/workflows/build-yocto.yml:4]: we can drop this PR if you want. Its not a big deal.

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**